libs/SalesforceSDKCore/SalesforceSDKCore/Classes/PushNotification/SFSDKPushNotificationEncryption.m

```diff
@@ -0,0 +1,40 @@
+/*
+Copyright (c) 2019-present, salesforce.com, inc. All rights reserved.
+
+Redistribution and use of this software in source and binary forms, with or without modification,
+are permitted provided that the following conditions are met:
+* Redistributions of source code must retain the above copyright notice, this list of conditions
+and the following disclaimer.
+* Redistributions in binary form must reproduce the above copyright notice, this list of
+conditions and the following disclaimer in the documentation and/or other materials provided
+with the distribution.
+* Neither the name of salesforce.com, inc. nor the names of its contributors may be used to
+endorse or promote products derived from this software without specific prior written
+permission of salesforce.com, inc.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR
+IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY
+WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+
+#import <SalesforceSDKCore/SFSDKCryptoUtils.h>
+#import "SFSDKPushNotificationEncryption.h"
+#import "SFSDKPushNotificationEncryptionConstants.h"
+
+@implementation SFSDKPushNotificationEncryption
+
++ (NSString *)getRSAPublicKey {
+    NSString *rsaPublicKey = [SFSDKCryptoUtils getRSAPublicKeyStringWithName:kPNEncryptionKeyName keyLength:kPNEncryptionKeyLength];
+    if (rsaPublicKey == nil) {
+        [SFSDKCryptoUtils createRSAKeyPairWithName:kPNEncryptionKeyName keyLength:kPNEncryptionKeyLength accessibleAttribute:kSecAttrAccessibleAfterFirstUnlock];
```

Comment:
Changed this from deprecated `kSecAttrAccessibleAlways`

Reply:
Should the app decide the accessible attribute to use? In other words, should `getRSAPublicKey` take a `accessibleAttribute` parameter?

Reply:
It will probably be flagged by security if we use `kSecAttrAccessibleAlways`. See https://github.com/forcedotcom/SalesforceMobileSDK-iOS/pull/2827

Reply:
I don't know about that. Devs might unknowingly change this to some more restrictive value and PNs will stop working suddenly. We see GitHub issues being raised for keychain issues due to incorrect values quite frequently. We might want to keep this internal to the SDK.

Reply:
Then I would use `kSecAttrAccessibleAfterFirstUnlockThisDeviceOnly`.

Reply:
Agreed. Sounds good 👍